src/main/java/com/durov/maks/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -52,4 +54,18 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("can't get all cinema hall entity", e);
         }
     }
+
+    @Override
+    public CinemaHall getById(Long cinemaHallId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<CinemaHall> criteriaQuery = criteriaBuilder.createQuery(CinemaHall.class);
+            Root<CinemaHall> cinemaHallRoot = criteriaQuery.from(CinemaHall.class);
+            criteriaQuery.select(cinemaHallRoot).where(criteriaBuilder
+                    .equal(cinemaHallRoot.get("id"), cinemaHallId));
+            return session.createQuery(criteriaQuery).getSingleResult();
```

Comment:
return session.get(CinemaHall.class, cinemaHallId);
try this one(here and everywhere where you are getting entity by id)

Reply:
got it